src/zarr/storage/remote.py

```diff
@@ -94,7 +94,11 @@ async def clear(self) -> None:
             pass
 
     async def empty(self) -> bool:
-        return not await self.fs._find(self.path, withdirs=True)
+        async for _path, _dirs, files in self.fs._walk(self.path):
```

Comment:
It cannot be guaranteed that this doesn't do a find() too... Would `_ls()` be enough?

There was talk of making `ls()` and friends in fsspec be generators, but not yet.

Reply:
gotcha... This was an improvement using gcsfs but I can see how it wouldn't be for others. I'll take your lead on the best path forward here. All we need to know is if there are any files in the store (ignoring empty directories).

Reply:
Given the very many files that can in theory be in a zarr hierarchy, maybe we can keep with this strategy for now. Having said that, in the zarr case, we do know which subdir ought to have the fewest items in it, right?